WordPress/gutenberg
# [RNMobile] Refactor Dark Mode HOC

## Description
Addresses feedbacks from https://github.com/WordPress/gutenberg/pull/17228
This PR fixes some antipattern in the naming of the HOC used for dark mode on gutenberg mobile.
List of changes:
- Moved from `@wordpress/components` to `@wordpress/compose`
- Renamed HOC from `withTheme` to `withPreferredColorScheme` and move it to  `packages/compose/higher-order/with-preferred-color-scheme`
- Renamed injected prop to compute styles from `useStyle` to `getStylesFromColorScheme`
- Add new hook `usePreferredColorScheme` in `packages/compose/hooks/use-preferred-color-scheme` that simply [exports](https://github.com/WordPress/gutenberg/pull/17552/files#diff-a028bc66e9dad38eeac7ce1197477dbdR11) `useDarkModeContext ` from [`react-native-dark-mode`](https://www.npmjs.com/package/react-native-dark-mode)
- Change [implementation](https://github.com/WordPress/gutenberg/pull/17552/files#diff-6a8479ece0ad578559d72f5f58e2bdf4R7) of the `withPreferredColorScheme` to reuse the `usePreferredColorScheme` hook
- Update references to `withTheme` and `useStyle` in native code

## How has this been tested?
- Try the demo app on Android and iOS
- Try the demo app using Xcode 11 (beta) on iOS 13
- Set the simulator/device to DarkMode
- Check that all colors looks good and there are no errors.
    - All blocks
    - Bottom-sheets (Add link - Image settings)
    - Block inserter
    - Unsupported/Validation error placeholders
    - HTML mode

## Types of changes





## Checklist:
